### PR TITLE
feat: support funds by chain

### PR DESCRIPTION
**Pull request by Tanya-atatakai**
## Types of changes

What types of changes does your code introduce?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

File: frontend/components/MainPage/header/AgentButton/AgentNotRunningButton.tsx [hunk 1]
```diff
@@ -127,13 +129,6 @@ const useServiceDeployment = () => {
...
     }
 
-    const masterSafeOlasBalance = masterSafeBalances?.find(
-      (walletBalanceResult) =>
-        walletBalanceResult.symbol === TokenSymbol.OLAS &&
-        walletBalanceResult.evmChainId === selectedAgentConfig.evmHomeChainId,
-    )?.balance;
-
-    const isInitialFunded = storeState?.[selectedAgentType]?.isInitialFunded;
     if (service && isInitialFunded && isServiceStaked) {
```
Comment (Tanya-atatakai) [inline frontend/components/MainPage/header/AgentButton/AgentNotRunningButton.tsx:132]: @truemiller can you please look at this code and the `" // SERVICE IS STAKED, AND STARTING AGAIN"` code section below and tell me if I'm right or not - it looks, that if a user runs the agent for the second time, the code will alway end up here, and no "threshold check" check will happen (to check if a user has enough gas for starting the agent)

also, here we check this: `(serviceTotalStakedOlas ?? 0) >= requiredStakedOlas` but if a user staked already, their total staked will be more than required anyway, what sense does this make?

File: frontend/config/stakingPrograms/mode.ts [hunk 2]
```diff
@@ -13,7 +13,7 @@ import { StakingProgramMap } from '.';
 export const MODE_STAKING_PROGRAMS_CONTRACT_ADDRESSES: Record<string, Address> =
   {
     [StakingProgramId.OptimusAlpha]:
-      '0x534c0a05b6d4d28d5f3630d6d74857b253cf8332',
+      '0x5fc25f50E96857373C64dC0eDb1AbCBEd4587e91',
```
Comment (Tanya-atatakai) [inline frontend/config/stakingPrograms/mode.ts:16]: this was pointing to wrong Modius alpha, updated to the right one

File: frontend/client/types.ts [hunk 3]
```diff
@@ -33,8 +33,10 @@ export type ChainData = {
     agent_id: number;
     cost_of_bond: number;
     fund_requirements: {
-      agent: number;
-      safe: number;
+      [tokenSymbol: string]: {
```
Comment (OjusWiZard) [inline frontend/client/types.ts:36]: it should be token address actually, or is it just for labelling?

File: .operate.zip
Comment (OjusWiZard) [inline .operate.zip:1]: this haha

File: frontend/client/types.ts (hunk 3 above)
Comment (Tanya-atatakai) [reply frontend/client/types.ts:36]: oh right good catch

File: .operate.zip
Comment (Tanya-atatakai) [reply .operate.zip:1]: lol.........=\

**Pull request merged**